paasta_tools/setup_crd.py

```diff
@@ -0,0 +1,250 @@
+        version='v1alpha1',
+        kube_kind=KubeKind(singular='FlinkCluster', plural='flinkclusters'),
+    ),
+]
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description='Creates custom_resources.')
+    parser.add_argument(
+        '-d', '--soa-dir', dest="soa_dir", metavar="SOA_DIR",
+        default=DEFAULT_SOA_DIR,
+        help="define a different soa config directory",
+    )
+    parser.add_argument(
+        '-v', '--verbose', action='store_true',
+        dest="verbose", default=False,
+    )
+    args = parser.parse_args()
+    return args
+
+
+def main() -> None:
+    args = parse_args()
+    soa_dir = args.soa_dir
+    if args.verbose:
+        logging.basicConfig(level=logging.DEBUG)
+    else:
+        logging.basicConfig(level=logging.WARNING)
+
+    kube_client = KubeClient()
+
+    ensure_paasta_namespace(kube_client)
+
+    setup_kube_succeeded = setup_all_custom_resources(
+        kube_client=kube_client,
+        soa_dir=soa_dir,
+        cluster=load_system_paasta_config().get_cluster(),
+    )
+    sys.exit(0 if setup_kube_succeeded else 1)
+
+
+def setup_all_custom_resources(
+    kube_client: KubeClient,
+    soa_dir: str,
+    cluster: str,
+) -> bool:
+    results = []
+    for custom_resource in CUSTOM_RESOURCES:
+        config_dicts = load_all_configs(
+            cluster=cluster,
+            file_prefix=custom_resource.file_prefix,
+            soa_dir=soa_dir,
+        )
+        results.append(
+            setup_custom_resources(
+                kube_client=kube_client,
+                kind=custom_resource.kube_kind,
+                config_dicts=config_dicts,
+                version=custom_resource.version,
+            ),
+        )
+    return all(results) if results else True
+
+
+def load_all_configs(cluster: str, file_prefix: str, soa_dir: str) -> Mapping[str, Mapping[str, Any]]:
+    config_dicts = {}
+    for service, _ in get_services_for_cluster(
+        cluster=cluster,
+        instance_type=file_prefix,
+        soa_dir=soa_dir,
+    ):
+        config_dicts[service] = service_configuration_lib.read_extra_service_information(
+            service,
+            f"{file_prefix}-{cluster}",
+            soa_dir=soa_dir,
+        )
+    return config_dicts
+
+
+def setup_custom_resources(
+    kube_client: KubeClient,
+    kind: KubeKind,
+    version: str,
+    config_dicts: Mapping[str, Mapping[str, Any]],
+) -> bool:
+    succeded = True
+    if config_dicts:
+        crs = list_all_custom_resources(
+            kube_client=kube_client,
+            plural_kind=kind.plural,
+            version=version,
+        )
+    for service, config in config_dicts.items():
+        if not reconcile_kubernetes_resource(
+            kube_client=kube_client,
+            service=service,
+            instance_configs=config,
+            kind=kind,
+            custom_resources=crs,
+            version=version,
+        ):
+            succeded = False
```

Comment:
Either `break` for `False` or collect results in a list and then `return all(...) ...`